### DISCUSSION TITLE
No FIELD_ADDED changes emitted for new types

### DISCUSSION THREAD

**Discussion by helfer** [kamilkisiela/graphql-inspector#1774] [General] [answered]
Hey @kamilkisiela, I noticed that when a new type is added, no FIELD_ADDED changes are emitted for the fields in that type. Is that intentional? I want to add a validation rule that marks `snake_case` fields as invalid, but without FIELD_ADDED events being emitted that's obviously not possible.

**Comment by kamilkisiela** [comment, marked as answer]
Not intentional but I will try today to include those changes. Thanks

**Reply by helfer** [reply]
Thanks @kamilkisiela , that would be awesome!

**Reply by kamilkisiela** [reply]
@helfer I started implementing it https://github.com/kamilkisiela/graphql-inspector/commit/4cb8d15bac1ae63a6630ea6c3911a08f3a7a6a0a and now I think showing "Type A was added" instead of its fields and their arguments, descriptions, default values is something expected and done by design. Otherwise where's the boundary for list of things to display when a new type is introduced.

I thought about implementing `full` mode as you can see in the commit but to be honest, maybe you need [graphql-eslint](https://the-guild.dev/blog/introducing-graphql-eslint) we introduced few weeks ago.

**Reply by kamilkisiela** [reply]
This way you could have a rule and run it on the entire schema
